src/lib/turtle.js

```diff
@@ -2028,18 +2046,35 @@ function generateTurtleModule(_target) {
         return result;
     }
 
-    function createColor(color, g, b, a) {
+    function createColor(turtle,color, g, b, a) {
         var i;
 
         if (g !== undefined) {
             color = [color, g, b, a];
         }
 
         if (color.constructor === Array && color.length) {
-            for(i = 0; i < 3; i++) {
-                color[i] = (typeof color[i] === "number") ?
-                    Math.max(0, Math.min(255, parseInt(color[i]))) :
-                    0;
+            if(turtle._colorMode === 255){//mode is 255
+                for(i = 0; i < 3; i++) {
+                    if(typeof color[i] === "number") {
+                        color[i] = Math.max(0, Math.min(255, parseInt(color[i])));
+                    } else { 
+                        throw new Sk.builtin.ValueError("bad color sequence");
+                    }
+                }
+            } else {//In python,if the colormode not equals 255,it should be 1.0
+                for(i = 0; i < 3; i++) {
+                    if(typeof color[i] === "number") {
+                        if(color[i] <= 1){
+                            color[i] = Math.max(0, Math.min(255, parseInt(255 * color[i])));
```

Comment:
I'd consider removing the `if/else` block and just replacing `color[i] = Math.max(0, Math.min(255, parseInt(255 * color[i])));` with `color[i] = Math.max(0, Math.min(255, parseInt(colorMode * color[i])));`.

Reply:
Yeah!Completed your two suggests directly in this branch.

Reply:
Because here should raise an error,so i didn't change it as you suggested.

Reply:
Didn't change for your suggestion about replace if/else.